src/components/Button/index.d.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Button props extending those of a regular button.
+ */
+export interface ButtonProps
```

Comment:
I'd say better to move this into the Button.tsx file. Declaration files should only be necessary for JS files. I suppose the build process has to then generate these